sklearn/utils/estimator_checks.py

```diff
@@ -878,15 +878,26 @@ def check_array_api_input(
     X_xp = xp.asarray(X, device=device)
     y_xp = xp.asarray(y, device=device)
 
-    est.fit(X, y)
+    if "Label" in est.__class__.__name__:
```

Comment:
The check functions are meant to be used outside of scikit-learn (e.g. by 3rd party libraries). Therefore we should not make any assumption on their names.